modular_skyrat/modules/paycheck_rations/code/quirk.dm

```diff
@@ -0,0 +1,69 @@
+/datum/quirk/item_quirk/ration_system
+	name = "Ration Ticket Receiver"
+	desc = "Due to some circumstance of your life, you have enrolled in the ration tickets program, \
+		which will halve all of your paychecks in exchange for granting you ration tickets, which can be \
+		redeemed at a cargo console for food and other items."
+	icon = FA_ICON_DONATE
+	quirk_flags = QUIRK_HUMAN_ONLY|QUIRK_HIDE_FROM_SCAN
+	value = 0
+	hardcore_value = 0
+
+/datum/quirk/item_quirk/ration_system/add_unique(client/client_source)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	if(!human_holder.account_id)
+		return
+	var/datum/bank_account/account = SSeconomy.bank_accounts_by_id["[human_holder.account_id]"]
+
+	var/obj/new_ticket_book = new /obj/item/storage/ration_ticket_book(get_turf(human_holder))
+	give_item_to_holder(
+		new_ticket_book,
+		list(
+			LOCATION_LPOCKET = ITEM_SLOT_LPOCKET,
+			LOCATION_RPOCKET = ITEM_SLOT_RPOCKET,
+			LOCATION_BACKPACK = ITEM_SLOT_BACKPACK,
+			LOCATION_HANDS = ITEM_SLOT_HANDS,
+		),
+	)
+	account.tracked_ticket_book = new_ticket_book
+	account.payday_modifier = 0.5
+	to_chat(client_source.mob, span_notice("You remember to keep close hold of your ticket book, it can't be replaced if lost and all of your ration tickets are placed there!"))
+
+// Edits to bank accounts to make the above possible
+
+/datum/bank_account
+	/// Tracks a linked ration ticket book. If we have one of these, then we'll put tickets in it every payday.
+	var/obj/item/storage/ration_ticket_book/tracked_ticket_book
```

Comment:
This will cause hard deletes if the ticket book gets deleted (explosion or w/e), you may want to use a weakref instead.